portal/src/main/webapp/js/src/pancancer_study_summary/pancancer_study_summary.js

```diff
@@ -790,12 +790,20 @@ function DataManagerPresenter(dmInitCallBack)
 			
 			//do the next call:
 			console.log(new Date() + ": CALL to get sample clinical atttributes (cancer types)");
-			return window.QuerySession.getSampleClinicalData(["CANCER_TYPE","CANCER_TYPE_DETAILED"]);
+			//this and next call below are in parallel, so the last one should call the dmInitCallBack:
+			nrResponsesReceived++;
+			if (nrResponsesReceived == 2) {
+				dmInitCallBack(self);
+			}
```

Comment:
hi Adam, thanks for the tip. I was aware of this option, however I chose to not do it like this since the function in then() will only be called when both genomic_data_fetched, clinical_data_fetched are ready. By splitting it up, I can now do the respective javascript processing of each part independently. So if e.g. genomic_data_fetched returns faster than clinical_data_fetched, I can process this data while the download of clinical_data_fetched continues. If there is a better way of doing this it would be great. I would hope there is something like:
```
$.when(data1, data2).then(functionA, functionB, functionC);
```
where functionA is called when data1 is in, functionB is called when data2 is in, and functionC is called when functionA and functionB are both done... I checked quickly but could not find something like this.